src/convert.test.js

```diff
@@ -0,0 +1,79 @@
+import JSBI from 'jsbi'
+import { getConvertedAmount } from './convert'
+import TokenAmount from './TokenAmount'
+
+const ONE_ETH = JSBI.BigInt('1000000000000000000')
+
+describe('getConvertedAmount tests', () => {
+  test('Converts amounts correctly', () => {
+    expect(getConvertedAmount(JSBI.BigInt('1'), 1).toString()).toEqual('1')
+    expect(getConvertedAmount(ONE_ETH, 1).toString()).toEqual(
+      ONE_ETH.toString()
+    )
+    expect(getConvertedAmount(JSBI.BigInt('1'), 0.5).toString()).toEqual('2')
+    expect(getConvertedAmount(JSBI.BigInt('1'), 0.25).toString()).toEqual('4')
+    expect(getConvertedAmount(JSBI.BigInt('1'), 0.125).toString()).toEqual('8')
+
+    expect(getConvertedAmount(JSBI.BigInt('100'), 50).toString()).toEqual('2')
+    // This is the exact case that broke the previous implementation,
+    // which is AAVE's amount of WBTC + the exchange rate at a certain
+    // hour on 2020-06-24
+    expect(
+      getConvertedAmount(JSBI.BigInt('1145054'), 0.00009248).toString()
+    ).toEqual('12381639273')
+  })
+
+  test('Converts amounts correctly with rates being strings', () => {
+    expect(getConvertedAmount(JSBI.BigInt('1'), '1').toString()).toEqual('1')
+    expect(getConvertedAmount(ONE_ETH, '1').toString()).toEqual(
+      ONE_ETH.toString()
+    )
+    expect(getConvertedAmount(JSBI.BigInt('1'), '0.5').toString()).toEqual('2')
+    expect(getConvertedAmount(JSBI.BigInt('1'), '0.25').toString()).toEqual('4')
+    expect(getConvertedAmount(JSBI.BigInt('1'), '0.125').toString()).toEqual(
+      '8'
+    )
+
+    expect(getConvertedAmount(JSBI.BigInt('100'), '50').toString()).toEqual('2')
+    // This is the exact case that broke the previous implementation,
+    // which is AAVE's amount of WBTC + the exchange rate at a certain
+    // hour on 2020-06-24
+    expect(
+      getConvertedAmount(JSBI.BigInt('1145054'), '0.00009248').toString()
+    ).toEqual('12381639273')
+  })
```

Comment:
Should we add some tests where the rate is higher than the amount? This would be common when converting into a higher-priced asset.

Reply:
Ah, just noticed that doing this actually [breaks the API](https://github.com/aragon/token-amount/pull/7/files#diff-17043324503bc20a641fb90473de9facR20-R24) due to not handling this case. Any ideas on how to handle this cleanly?